src/runtime/type.ts

```diff
@@ -134,7 +134,7 @@ export abstract class Type {
   }
 
   collectionOf() {
-    return new CollectionType(this);
+    return new CollectionType<Type>(this);
```

Comment:
The type will vary based on the `this` value.  So you should not include the explicit <Type> here.